docs/DeveloperGuide.md

```diff
@@ -1,29 +1,173 @@
 # Developer Guide
 
-## Acknowledgements
+## Design & Implementation
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+### Ui
+The `Ui` class handles user input and basic output messages. It can
+* Read user input
+* Output various messages to the user
 
-## Design & implementation
+![img.png](imgs/UiClass.png)
+
+### Storage Class
+![img.png](imgs/StorageClass.png)
+
+### Media Classes
+![img.png](imgs/MovieClass.png)
+
+### Commands
+The command component enables users to make changes to their review list. The command word is taken from the first word
+of the user input, and is processed through the `Parser` class. The class diagram shows how the Commands parent class is
+implemented, as well as its extended classes.
+![img.png](imgs/CommandsClass.png)
```

Comment:
![image](https://user-images.githubusercontent.com/53231968/198247957-03528b5e-2f56-406b-85ec-e72c37ce204e.png)

Might this italicised `String execute` be a bug?